src/class/msc/msc_device.c

```diff
@@ -753,6 +753,7 @@ static int32_t proc_builtin_scsi(uint8_t lun, uint8_t const scsi_cmd[16], uint8_
           .is_removable         = 1,
           .version              = 2,
           .response_data_format = 2,
+          .additional_length    = sizeof(scsi_inquiry_resp_t) - 4,
```

Comment:
This is `- 5` in our STM32 USB MSC driver... which one is correct?

Reply:
It's `- 4` in the both the SCSI reference manual linked above, and the RPI bootloader you mentioned.

Reply:
I realized this actually makes sense and it must be `- 4` because if it was `- 5` the kernel would have issued a warning (total length would have been 35). With `- 5` in the STM32 driver, it seems you're sending 37 bytes.

Reply:
I was wrong, I did some more testing and even though the specs say it's`N - 4`, I can confirm it should be `N - 5`. I used `sg_inq` util to send a standard inquiry, and with `N - 4` I get the following response (Note the length):
```
> sudo sg_inq --only /dev/sda
standard INQUIRY:
  PQual=0  PDT=0  RMB=1  LU_CONG=0  hot_pluggable=0  version=0x02  [SCSI-2]
  [AERC=0]  [TrmTsk=0]  NormACA=0  HiSUP=0  Resp_data_format=2
  SCCS=0  ACC=0  TPGS=0  3PC=0  Protect=0  [BQue=0]
  EncServ=0  MultiP=0  [MChngr=0]  [ACKREQQ=0]  Addr16=0
  [RelAdr=0]  WBus16=0  Sync=0  [Linked=0]  [TranDis=0]  CmdQue=0
    length=37 (0x25), but only fetched 36 bytes   Peripheral device type: disk
```
With `N - 5` I get the following response (36 bytes):
```
sudo sg_inq --only /dev/sda
standard INQUIRY:
  PQual=0  PDT=0  RMB=1  LU_CONG=0  hot_pluggable=0  version=0x02  [SCSI-2]
  [AERC=0]  [TrmTsk=0]  NormACA=0  HiSUP=0  Resp_data_format=2
  SCCS=0  ACC=0  TPGS=0  3PC=0  Protect=0  [BQue=0]
  EncServ=0  MultiP=0  [MChngr=0]  [ACKREQQ=0]  Addr16=0
  [RelAdr=0]  WBus16=0  Sync=0  [Linked=0]  [TranDis=0]  CmdQue=0
    length=36 (0x24)   Peripheral device type: disk
 Vendor identification: Micropy
 Product identification: Mass Storage
 Product revision level: 1.0
```

Reply:
Good to get to the bottom of this, and thanks for the hint about `sg_inq`.  I can see that the RP2040 bootloader has the same issue, a length of 37.

Reply:
See https://github.com/raspberrypi/pico-bootrom/issues/14

Reply:
>>  I can see that the RP2040 bootloader has the same issue, a length of 37.

Yes it seems to be a common issue due to the specs typo.